src/ext/h2c.js

```diff
@@ -34,43 +34,43 @@ function h2Base(x, curve, label) {
 /**
  * hashes bits to the chosen elliptic curve
  * @param {sjcl.bitArray} alpha bits to be encoded onto curve
- * @param {sjcl.ecc.curve} curve elliptic curve
- * @param {int} mode indicates which algorithm mode should be used (if any)
+ * @param {literal} ecSettings the curve settings being used by the extension
  * @return {sjcl.ecc.point} point on curve
  */
-function h2Curve(alpha, curve, mode) {
-  const curveName = sjcl.ecc.curveName(curve);
+function h2Curve(alpha, ecSettings) {
   let point;
-  switch (curveName) {
-    case "c256":
-      point = simplifiedSWU(alpha, mode);
+  switch (ecSettings.method) {
+    case "swu":
+      point = simplifiedSWU(alpha, ecSettings.curve, ecSettings.hash, SWU_MODE);
+      break;
+    case "increment":
+      point = hashAndInc(alpha, ecSettings.curve, ecSettings.hash);
       break;
     default:
-      throw new Error("[privacy-pass]: Incompatible curve chosen for hashing: " + curveName);
+      throw new Error("[privacy-pass]: Incompatible curve chosen for hashing: " + sjcl.ecc.curveName(ecSettings.curve));
```

Comment:
This line will show `c256` instead of `p256`

Reply:
I think this is unavoidable without rewriting the naming function, I will add "SJCL curve: " to the string to make it clear what we mean.